cmk/base/plugins/agent_based/cisco_asa_sensors.py

```diff
@@ -0,0 +1,340 @@
+# .1.3.6.1.2.1.99.1.1.1.6.15 = STRING: "celsius"
+# .1.3.6.1.2.1.99.1.1.1.6.16 = STRING: "celsius"
+
+#
+# sample string_table
+# [
+#  [
+#   ['Chassis', '', '', '', ''],
+#   ['Processor 0/0', '', '', '', ''],
+#   ['Processor 0/1', '', '', '', ''],
+#   ['Processor 0/2', '', '', '', ''],
+#   ['ASA5515 Slot for Removable Drive 0', '', '', '', ''],
+#   ['Micron_M550_MTFDDAK128MAY Removable Drive in Slot 0', '', '', '', ''],
+#   ['Chassis Cooling Fan 1', '', '', '', ''],
+#   ['Chassis Fan Sensor 1', '10', '7680', '1', 'rpm'],
+#   ['Chassis Cooling Fan 2', '', '', '', ''],
+#   ['Chassis Fan Sensor 2', '10', '7936', '1', 'rpm'],
+#   ['Chassis Cooling Fan 3', '', '', '', ''],
+#   ['Chassis Fan Sensor 3', '10', '7680', '1', 'rpm'],
+#   ['CPU Temperature Sensor 0/0', '8', '34', '1', 'celsius'],
+#   ['Chassis Ambient Temperature Sensor 1', '8', '32', '1', 'celsius'],
+#   ['Chassis Ambient Temperature Sensor 2', '8', '30', '1', 'celsius'],
+#   ['Chassis Ambient Temperature Sensor 3', '8', '33', '1', 'celsius'],
+#   ['Gi0/0', '', '', '', ''],
+#   ['Gi0/1', '', '', '', ''],
+#   ['Gi0/2', '', '', '', ''],
+#   ['Gi0/3', '', '', '', ''],
+#   ['Gi0/4', '', '', '', ''],
+#   ['Gi0/5', '', '', '', ''],
+#   ['In0/0', '', '', '', ''],
+#   ['In0/1', '', '', '', ''],
+#   ['Ma0/0', '', '', '', ''],
+#   ['Po1', '', '', '', '']
+#  ]
+# ]
+#
+# sample section
+# {'fan': {
+#    'Chassis Sensor 1': CiscoAsaSensor(value=7680, status=<State.OK: 0>, state_readable='Ok', unit='rpm'),
+#    'Chassis Sensor 2': CiscoAsaSensor(value=7936, status=<State.OK: 0>, state_readable='Ok', unit='rpm'),
+#    'Chassis Sensor 3': CiscoAsaSensor(value=7680, status=<State.OK: 0>, state_readable='Ok', unit='rpm')
+#   },
+#  'temp': {
+#    'CPU Sensor 0/0': CiscoAsaSensor(value=34.0, status=<State.OK: 0>, state_readable='Ok', unit='celsius'),
+#    'Chassis Ambient Sensor 1': CiscoAsaSensor(value=32.0, status=<State.OK: 0>, state_readable='Ok', unit='celsius'),
+#    'Chassis Ambient Sensor 2': CiscoAsaSensor(value=30.0, status=<State.OK: 0>, state_readable='Ok', unit='celsius'),
+#    'Chassis Ambient Sensor 3': CiscoAsaSensor(value=33.0, status=<State.OK: 0>, state_readable='Ok', unit='celsius')
+#   },
+#  'power': {}}
+#
+
+from typing import Dict, List, NamedTuple
+
+from .agent_based_api.v1.type_defs import (
+    DiscoveryResult,
+    CheckResult,
+    StringTable,
+)
+
+from .agent_based_api.v1 import (
+    register,
+    Service,
+    Result,
+    check_levels,
+    State,
+    SNMPTree,
+    startswith,
+)
+
+
+# ##################################################################################################
+#
+# ASA TEMPERATURE BASE
+#
+# ##################################################################################################
+
+
+class CiscoAsaSensor(NamedTuple):
+    value: float
+    status: State
+    state_readable: str
+    unit: str
+
+
+def parse_cisco_asa_sensors(string_table: List[StringTable]) -> Dict:
```

Comment:
The return type can be improved. You can for example use a NamedTuple with the fields `temp`, `fan` and `power`, each of the type `Mapping[str, CiscoAsaSensor]`